cmd/loki/main.go

```diff
@@ -51,12 +51,18 @@ func main() {
 	util.InitLogger(&config.Server)
 
 	// Setting the environment variable JAEGER_AGENT_HOST enables tracing
-	trace := tracing.NewFromEnv(fmt.Sprintf("loki-%s", config.Target))
+	trace, err := tracing.NewFromEnv(fmt.Sprintf("loki-%s", config.Target))
+	if err != nil {
+		level.Error(util.Logger).Log("msg", "error in initializing tracing. tracing will not be enabled", "err", err)
+	}
 	defer func() {
-		if err := trace.Close(); err != nil {
-			level.Error(util.Logger).Log("msg", "error closing tracing", "err", err)
-			os.Exit(1)
+		if trace != nil {
+			if err := trace.Close(); err != nil {
+				level.Error(util.Logger).Log("msg", "error closing tracing", "err", err)
+				os.Exit(1)
```

Comment:
I don't think we need this os.Exit(1) I know it was there but I'm willing to get rid of it.

Reply:
removed

Reply:
There is some go mod sync issue still.